SU2_CFD/src/output_structure.cpp

```diff
@@ -18041,6 +18062,398 @@ void COutput::DeallocateSurfaceData_Parallel(CConfig *config, CGeometry *geometr
+
+  unsigned short nVar_Turb = 0;
+  if (turbulent)
+    switch (config->GetKind_Turb_Model()) {
+      case SA: case SA_NEG: case SA_E: case SA_COMP: case SA_E_COMP:
+        nVar_Turb = 1;
+        turb_val[0] = solver[TURB_SOL]->GetNuTilde_Inf();
+        break;
+      case SST:
+        nVar_Turb = 2;
+        turb_val[0] = solver[TURB_SOL]->GetTke_Inf();
+        turb_val[1] = solver[TURB_SOL]->GetOmega_Inf();
+        break;
+      default:
+        SU2_MPI::Error("Specified turbulence model unavailable or none selected", CURRENT_FUNCTION);
+        break;
+    }
+
+  /*--- Count the number of columns that we have for this flow case.
+   Here, we have nDim entries for node coordinates, 2 entries for the total
+   conditions or mass flow, another nDim for the direction vector, and
+   finally entries for the number of turbulence variables. ---*/
+
+  unsigned short nCol_InletFile = nDim + 2 + nDim + nVar_Turb;
+
+  /*--- Write the inlet profile file. Note that we have already merged
+   all of the information for the markers and coordinates previously
+   in the MergeInletCoordinates() routine. ---*/
+
+  ofstream node_file("inlet_example.dat");
+
+  node_file << "NMARK= " << nMarker_InletFile << endl;
+
+  for (iMarker = 0; iMarker < nMarker_InletFile; iMarker++) {
+
+    /*--- Access the default data for this marker. ---*/
+
+    string Marker_Tag   = Marker_Tags_InletFile[iMarker];
+    su2double p_total   = config->GetInlet_Ptotal(Marker_Tag);
+    su2double t_total   = config->GetInlet_Ttotal(Marker_Tag);
+    su2double* flow_dir = config->GetInlet_FlowDir(Marker_Tag);
+
+    /*--- Header information for this marker. ---*/
+
+    node_file << "MARKER_TAG= " << Marker_Tag              << endl;
+    node_file << "NROW="        << nRow_InletFile[iMarker] << endl;
+    node_file << "NCOL="        << nCol_InletFile          << endl;
+
+    node_file << setprecision(15);
+    node_file << std::scientific;
+
+    /*--- Loop over the data structure and write the coords and vars to file. ---*/
+
+    for (iPoint = nRowCum_InletFile[iMarker]; iPoint < nRowCum_InletFile[iMarker+1]; iPoint++) {
+
+      for (iDim = 0; iDim < nDim; iDim++) {
+        node_file << InletCoords[iDim][iPoint] << "\t";
+      }
+      node_file << t_total << "\t" << p_total;
+      for (iDim = 0; iDim < nDim; iDim++) {
+        node_file << "\t" << flow_dir[iDim];
+      }
+      for (iVar = 0; iVar < nVar_Turb; iVar++) {
+        node_file << "\t" << turb_val[iVar];
+      }
+      node_file << endl;
+    }
+
+  }
+  node_file.close();
+
+  /*--- Print a message to inform the user about the template file. ---*/
+
+  cout << endl;
+  cout << "  Created a template inlet profile file with node coordinates" << endl;
+  cout << "  and solver variables at `inlet_example.dat`." << endl;
+  cout << "  You can use this file as a guide for making your own inlet" << endl;
+  cout << "  specification." << endl << endl;
+  
```

Comment:
Do you think that it is better to continue the calculation or just abort if the code does not find the inlet profile?

Reply:
Good point. This might confuse users into thinking that the calculation completed successfully without even noticing that the profile was not used. I will update this to throw an error instead. We'll merge once the tests pass after that.